src/lib/raft/raft.c

```diff
@@ -367,6 +408,30 @@ raft_process_recovery(struct raft *raft, const struct raft_msg *req)
 	assert(!raft_is_enabled(raft));
 }
 
+/** Update raft state once leader is seen. */
+static bool
+raft_leader_see(struct raft *raft);
```

Comment:
Perhaps the announcement is not needed anymore:
```Diff
@@ -410,7 +410,18 @@ raft_process_recovery(struct raft *raft, const struct raft_msg *req)
 
 /** Update raft state once leader is seen. */
 static bool
-raft_leader_see(struct raft *raft);
+raft_leader_see(struct raft *raft)
+{
+	uint32_t source = raft->self;
+	assert(source > 0 && source < VCLOCK_MAX);
+	bool was_seen = bit_set(&raft->leader_witness_map, source);
+	raft->leader_last_seen = raft_ev_monotonic_now(raft_loop());
+	if (!was_seen) {
+		raft_schedule_broadcast(raft);
+		return true;
+	}
+	return false;
+}
 
 /**
  * Clear leader-related fields on leader loss or resign and let everyone know
@@ -1104,20 +1115,6 @@ raft_notify_is_leader_seen(struct raft *raft, bool is_leader_seen,
 	return was_seen != is_leader_seen;
 }
 
-static bool
-raft_leader_see(struct raft *raft)
-{
-	uint32_t source = raft->self;
-	assert(source > 0 && source < VCLOCK_MAX);
-	bool was_seen = bit_set(&raft->leader_witness_map, source);
-	raft->leader_last_seen = raft_ev_monotonic_now(raft_loop());
-	if (!was_seen) {
-		raft_schedule_broadcast(raft);
-		return true;
-	}
-	return false;
-}
```

Reply:
Ok, I moved it together with `raft_notify_is_leader_seen`, because I want to keep these two functions close.
```diff
diff --git a/src/lib/raft/raft.c b/src/lib/raft/raft.c
index 7a19bed99..0fd77edb1 100644
--- a/src/lib/raft/raft.c
+++ b/src/lib/raft/raft.c
@@ -408,9 +408,40 @@ raft_process_recovery(struct raft *raft, const struct raft_msg *req)
 	assert(!raft_is_enabled(raft));
 }
 
+bool
+raft_notify_is_leader_seen(struct raft *raft, bool is_leader_seen,
+			   uint32_t source)
+{
+	assert(source > 0 && source < VCLOCK_MAX && source != raft->self);
+	/* Leader doesn't care whether someone sees it or not. */
+	if (raft->state == RAFT_STATE_LEADER)
+		return false;
+
+	bool was_seen;
+	if (is_leader_seen) {
+		was_seen = bit_set(&raft->leader_witness_map, source);
+	} else {
+		was_seen = bit_clear(&raft->leader_witness_map, source);
+		if (was_seen)
+			raft_sm_election_update(raft);
+	}
+	return was_seen != is_leader_seen;
+}
+
 /** Update raft state once leader is seen. */
 static bool
-raft_leader_see(struct raft *raft);
+raft_leader_see(struct raft *raft)
+{
+	uint32_t source = raft->self;
+	assert(source > 0 && source < VCLOCK_MAX);
+	bool was_seen = bit_set(&raft->leader_witness_map, source);
+	raft->leader_last_seen = raft_ev_monotonic_now(raft_loop());
+	if (!was_seen) {
+		raft_schedule_broadcast(raft);
+		return true;
+	}
+	return false;
+}
 
 /**
  * Clear leader-related fields on leader loss or resign and let everyone know
@@ -1082,40 +1113,6 @@ raft_cfg_is_candidate(struct raft *raft, bool is_candidate)
 	raft_restore(raft);
 }
 
-bool
-raft_notify_is_leader_seen(struct raft *raft, bool is_leader_seen,
-			   uint32_t source)
-{
-	assert(source > 0 && source < VCLOCK_MAX && source != raft->self);
-	/* Leader doesn't care whether someone sees it or not. */
-	if (raft->state == RAFT_STATE_LEADER)
-		return false;
-
-	bool was_seen;
-	if (is_leader_seen) {
-		was_seen = bit_set(&raft->leader_witness_map, source);
-	} else {
-		was_seen = bit_clear(&raft->leader_witness_map, source);
-		if (was_seen)
-			raft_sm_election_update(raft);
-	}
-	return was_seen != is_leader_seen;
-}
-
-static bool
-raft_leader_see(struct raft *raft)
-{
-	uint32_t source = raft->self;
-	assert(source > 0 && source < VCLOCK_MAX);
-	bool was_seen = bit_set(&raft->leader_witness_map, source);
-	raft->leader_last_seen = raft_ev_monotonic_now(raft_loop());
-	if (!was_seen) {
-		raft_schedule_broadcast(raft);
-		return true;
-	}
-	return false;
-}
-
 void
 raft_cfg_is_candidate_later(struct raft *raft, bool is_candidate)
 {

```